server/src/main/java/org/solarbank/server/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,60 @@
+package org.solarbank.server;
+
+import javax.money.UnknownCurrencyException;
+import org.solarbank.server.Messages;
+import org.solarbank.server.dto.ErrorResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class ControllerExceptionHandler {
```

Comment:
Can we add unit tests for the handlers which:
1.  pass in an exception
3. assert that the ResponseEntity is as expected

The validation handler at least needs tests as it has logic for formatting the response messages.

Reply:
added unit tests for the exception handlers